### PR TITLE
Avoid UTF8->UTF16->UTF8 roundtrip when logging on mobile

### PR DESCRIPTION
**Pull request by bstoeger**
Make writeToAppLogFile() take an std::string parameter to avoid an unnecessary UTF8->UTF16->UTF8 (std::string->QString->char *) roundtrip.

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Tiny code cleanup that avoids unnecessary UTF8->UTF16->UTF8 conversions.

### Related issues:

Follow up to #4385.

### CODE REVIEW

**Comment by bstoeger** [comment]
@jefdriesen: I would appreciate your opinion, since I don't use mobile. Thanks.

**Review by mikeller — APPROVED** [review]

File: mobile-widgets/qmlmanager.cpp [hunk 1]
```diff
@@ -1849,20 +1849,20 @@ void QMLManager::setBtEnabled(bool value)
...
 	QMLManager *self = QMLManager::instance();
 	if (self) {
 		self->writeToAppLogFile(logText);
 	}
 }
 
-void QMLManager::writeToAppLogFile(QString logText)
+void QMLManager::writeToAppLogFile(const std::string &logText)
 {
+	std::string line = logText + "\n";
```
Comment (jefdriesen) [inline mobile-widgets/qmlmanager.cpp]: The appending of the new line could be moved inside the if statement, such that you only have to copy the string when it's actually written to the log.

**Comment by jefdriesen** [comment]
Looks good to me. The fewer conversions (and copies) the better.

File: mobile-widgets/qmlmanager.cpp (hunk 1 above)
Comment (bstoeger) [reply mobile-widgets/qmlmanager.cpp]: Good point.

**Pull request merged**